crates/cubecl-opt/src/block.rs

```diff
@@ -0,0 +1,53 @@
+use std::{cell::RefCell, collections::HashSet, rc::Rc};
+
+use cubecl_core::ir::{Operation, Variable};
+use petgraph::graph::NodeIndex;
+use stable_vec::StableVec;
+
+use crate::{version::PhiInstruction, ControlFlow, Optimizer, Program};
+
+#[derive(Default, Debug, Clone)]
+pub struct BasicBlock {
```

Comment:
I think it's important to add documentation on the new data structures added to the new crate.